### ISSUE TITLE
filters on postobject may not work as expected. global post object is not setup.

### ISSUE THREAD

**Issue by jespervnielsen** [wp-graphql/wp-graphql#718]
https://github.com/wp-graphql/wp-graphql/blob/develop/src/Type/Object/PostObject.php#L267

```apply_filters( 'the_excerpt', $excerpt );```
this is in a way correct. 
But since the global post object isnt't setup, then there may be scenarios, where it will be the wrong post.

a way to solve this, could be to use
```
   global $post;
   $post = $post_object;
   
   setup_postdata( $post );
$excerpt = apply_filters( 'get_the_excerpt', $excerpt, $post );
$excerpt =apply_filters( 'the_excerpt', $excerpt );
wp_reset_postdata();
return $excerpt;
```

The same issue is relevant for "the_content" and "the_title"

**Comment by epeli** [comment]
I believe this will be fixed by

https://github.com/wp-graphql/wp-graphql/pull/714

Care to try manually merging it and checking if it fixes it for you?

**Comment by jespervnielsen** [comment]
@epeli i can confirm that #714 solves this. 

i did a checkout of 714, and the issues is solved.

And i makes sense to fill the global post object, on a field level, as the PR does.

**Comment by epeli** [comment]
Thanks. I hope we can get it merged soon.

In the mean time if you don't want to run forked version you should be able to do:

```php
add_action('graphql_before_resolve_field', function($source) {
  if (is_a( $source, 'WP_Post' ) ) {
    $GLOBALS['post'] = $source;
    setup_postdata( $source );
  }
} );

```

**Comment by jasonbahl** [comment]
@jespervnielsen @epeli reviewing the PR now. Hoping to merge today. 😄

**Comment by epeli** [comment]
Merged 💪

**Comment by jasonbahl** [comment]
closed by #714
